sdv/tabular/base.py

```diff
@@ -457,7 +457,7 @@ def sample(self, num_rows=None, max_retries=100, max_rows_multiplier=10,
 
             condition_indices = dataframe[COND_IDX]
             condition = dict(zip(condition_columns, group))
-            if transformed_conditions.empty:
+            if len(transformed_columns) == 0:
```

Comment:
Interesting point here. The `transformed_conditions.empty` was actually covering the case where there are no rows in the `DataFrame`, rather than the case where there are no columns (since there always was at least the COND_IDX column).
But since we are inside the `groupby` loop, we know that there is at least 1 `conditions` row, which means that the `transformed_conditions` cannot be empty!
So, if I'm reading it right, this block of code was unreachable in the previous version!

Reply:
So, to add to the previous comment: Can we delete this test?
https://github.com/sdv-dev/SDV/blob/0d4163ca537f22bb47385badb9e1399fc1f72928/tests/unit/tabular/test_base.py#L85

To be fair, the current unit tests for the `BaseTabularModel` are not properly implemented, since they should be done inside a `class` called `TestBaseTabularModel` which would implement at least 1 test for each one of the methods inside the `BaseTabularModel`.

Can you get started on creating the test class and adding at least the `no transformed_columns` unit test for the `sample` method? The thing should be similar to what can be found in https://github.com/sdv-dev/SDV/blob/master/tests/unit/tabular/test_copulas.py

Reply:
I just raised issue #543 to cover the proper testing of the `BaseTabularModel`. That will be covered in a separated PR, but let's get started on adding at least the test suggested above to the new test class.

Reply:
Yes, I believe it was previously unreachable because we create the `COND_IDX` column in `transformed_conditions` beforehand.

I removed the unneeded test, created the class `TestBaseTabularModel`, and added a `no_transformed_columns` unit test. Moving the existing test cases in a separate issue sounds good to me.

Reply:
This is strange. I remember adding the test for this because of an issue I ran into while testing it, but I can't remember what it was

Reply:
When looking at this again, I found that if you run `df.empty` on a `DataFrame` that has no columns/rows but does have an index, it still returns `True`. So I'm confused how this case wasn't being reached

Reply:
Sorry I was confused because I forgot that `transformed_conditions.reset_index(inplace=True)`, will move the previous index column into the `DataFrame`